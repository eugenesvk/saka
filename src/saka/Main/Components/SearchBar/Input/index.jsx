import { Component, h } from 'preact';

export default class Input extends Component {
  render() {
    const {
      placeholder,
      searchString,
      onKeyDown,
      onInput,
      onBlur
    } = this.props;

    return (
      <section className="mdc-text-field mdc-text-field--filled mdc-text-field--fullwidth mdc-text-field--no-label search-field-wrapper">
        <input
          id="search-bar"
          className="mdc-text-field__input search-field-input"
          type="text"
          placeholder={placeholder}
          aria-label={placeholder}
          onKeyDown={onKeyDown}
          onInput={onInput}
          value={searchString}
          onBlur={onBlur}
          ref={input => input && input.focus()}
        />
      </section>
    );
  }
}
