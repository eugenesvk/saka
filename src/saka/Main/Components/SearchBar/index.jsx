import { h } from 'preact';
import 'scss/styles.scss';
import Input from './Input/index.jsx';

export default ({
  placeholder,
  searchString,
  suggestion,
  onKeyDown,
  onInput,
  onBlur
}) => (
  <form className="search-bar-container">
    <Input
      placeholder={placeholder}
      searchString={searchString}
      suggestion={suggestion}
      onKeyDown={onKeyDown}
      onInput={onInput}
      onBlur={onBlur}
    />
  </form>
);
