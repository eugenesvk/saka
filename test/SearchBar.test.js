import { h } from 'preact';
import { render, cleanup, flushPromises } from '@testing-library/preact';
import SearchBar from '@/saka/Main/Components/SearchBar/index';

afterEach(cleanup);

test('should be empty when no there is no search string provided', async () => {
  const props = {
    placeholder: 'Tabs',
    searchString: '',
    suggestion: {},
    mode: 'tab',
    onKeyDown() {},
    onInput() {},
    onBlur() {},
    onButtonClick() {}
  };

  const { container } = render(<SearchBar {...props} />);
  expect(container).toMatchSnapshot();
});

test('should show the search string when search string is provided', async () => {
  const props = {
    placeholder: 'Tabs',
    searchString: 'Saka github',
    suggestion: {},
    mode: 'tab',
    onKeyDown() {},
    onInput() {},
    onBlur() {},
    onButtonClick() {}
  };

  const { getByPlaceholderText } = render(<SearchBar {...props} />);

  expect(getByPlaceholderText('Tabs').value).toBe('Saka github');
});
