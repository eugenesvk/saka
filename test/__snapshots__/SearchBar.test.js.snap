// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be empty when no there is no search string provided 1`] = `
<div>
  <form
    class="search-bar-container"
  >
    <section
      class="mdc-text-field mdc-text-field--filled mdc-text-field--fullwidth mdc-text-field--no-label search-field-wrapper"
    >
      <input
        aria-label="Tabs"
        class="mdc-text-field__input search-field-input"
        id="search-bar"
        placeholder="Tabs"
        type="text"
      />
    </section>
  </form>
</div>
`;
